File: ARCADE/The Core/Level 4_2: countSumOfTwoRepresentations2.cs
    using System;
     
    public class Test
    {
    	public static int countSumOfTwoRepresentations2(int n, int l, int r){
            int a = 0, b = 0, count = 0;
            if (n - l > r) { a = n - r; b = r; }
            else { a = l; b = n - a; }
            while (a <= b && b <= r){
                count++;
                a++;
                b--;
            }
            return count;
        }
    	public static void Main()
    	{
    		int n = 6, l = 2, r = 4;
    		Console.Write(countSumOfTwoRepresentations2(n, l, r));
    	}
    }

File: ARCADE/The Core/Level 4_2: countSumOfTwoRepresentations2.pas
program countSumOfTwoRepresentations2;

function countSumOfTwoRepresentations2(n: integer; l: integer; r: integer): integer;
var a, b, count : integer;
begin
    a := 0; b := 0; count := 0;
    if(n - l > r) then
    begin
        a := n - r;
        b := r;
    end
    else
    begin
        a := l;
        b := n - a;
    end;
    
    while((a <= b) and (b <= r)) do
    begin
        inc(count);
        inc(a);
        dec(b);
    end;
    
    countSumOfTwoRepresentations2 := count;
end;

var n, l, r : integer;
begin
	n := 6; l := 2; r := 4;
	writeln(countSumOfTwoRepresentations2(n, l, r));
end.
